modules/features/engage/src/main/kotlin/au/com/shiftyjelly/pocketcasts/engage/EngageSdkBridge.kt

```diff
@@ -0,0 +1,28 @@
+package au.com.shiftyjelly.pocketcasts.engage
+
+import android.content.Context
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.Feature
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.FeatureFlag
+import dagger.hilt.android.qualifiers.ApplicationContext
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class EngageSdkBridge @Inject constructor(
+    @ApplicationContext private val context: Context,
+) {
+    private val isEngageIntegrationEnabled = AtomicBoolean(false)
+
+    fun registerIntegration() {
+        if (FeatureFlag.isEnabled(Feature.ENGAGE_SDK) && !isEngageIntegrationEnabled.getAndSet(true)) {
```

Comment:
I'm not unregistering things with the feature disabled because I don't anticipate using the flag in production. I only added it here in case I don't manage to finish the integration by the next release.